py_cui/renderer.py

```diff
@@ -384,20 +387,21 @@ def draw_text(self, ui_element, line, y, centered = False, bordered = True, sele
 
         # Each text elem is a list with [text, color]
         for text_elem in render_text:
-            text_elem[0] = text_elem[0].replace(chr(0), "")
-            self.set_color_mode(text_elem[1])
+            if isinstance(text_elem[0],str) and isinstance(text_elem[1],int):
+                text_elem[0] = text_elem[0].replace(chr(0), "")
+                self.set_color_mode(text_elem[1])
 
-            # BLACK_ON_WHITE + BOLD is unreadable on windows terminals
-            if selected and text_elem[1] != py_cui.BLACK_ON_WHITE:
-                self._set_bold()
+                # BLACK_ON_WHITE + BOLD is unreadable on windows terminals
+                if selected and text_elem[1] != py_cui.BLACK_ON_WHITE:
+                    self._set_bold()
 
-            self._stdscr.addstr(y, current_start_x, text_elem[0])
-            current_start_x = current_start_x + len(text_elem[0])
+                self._stdscr.addstr(y, current_start_x, text_elem[0])
+                current_start_x = current_start_x + len(text_elem[0])
 
-            if selected and text_elem[1] != py_cui.BLACK_ON_WHITE:
-                self._unset_bold()
-            
-            self.unset_color_mode(text_elem[1])
+                if selected and text_elem[1] != py_cui.BLACK_ON_WHITE:
+                    self._unset_bold()
+                
+                self.unset_color_mode(text_elem[1])
```

Comment:
Is there a case where `text_elem` wouldn't be of type `List[str, int]`?

Reply:
I couldn't find any. I referred `def _generate_text_color_fragments(..)` method for this.